encoders/firebase-decoders-json/src/main/java/com/google/firebase/decoders/TypeToken.java

```diff
@@ -180,7 +181,8 @@ String getTypeTokenLiteral() {
   public static class ArrayToken<T> extends TypeToken<T> {
     private final TypeToken<?> componentType;
 
-    private ArrayToken(TypeToken<?> componentType) {
+    private ArrayToken(Class<T> rawType, TypeToken<?> componentType) {
+      super.rawType = rawType;
       this.componentType = componentType;
     }
```

Comment:
Consider removing the second argument and instead doing something like:

```suggestion
    private ArrayToken(Class<T> rawType) {
      if(!rawType.isArray()) {
        throw new IllegalArgumentException("blah-blah");
      }
      super(rawType);
      this.componentType = TypeToken.of(rawType.getComponentType());
    }
```